mypyc/test/test_emitfunc.py

```diff
@@ -245,6 +247,10 @@ def test_dict_contains(self) -> None:
             'in', self.b, self.o, self.d,
             """cpy_r_r0 = PyDict_Contains(cpy_r_d, cpy_r_o);""")
 
+    def test_binary_int_op(self) -> None:
+        self.assert_emit(BinaryIntOp(bool_rprimitive, self.s1, self.s2, BinaryIntOp.SLT, 1),
+                         """cpy_r_r0 = (Py_ssize_t)cpy_r_s1 < (Py_ssize_t)cpy_r_s2;""")
```

Comment:
Also test unsigned comparison with `int64` and `int32`. These need a cast to the corresponding unsigned type (`uint64_t` or `uint32_t`, respectively). Finally, test that unsigned comparison with `short_int_rprimitive` and signed comparison with `int32` or `int64` doesn't use a cast.

Reply:
Thanks! This actually catches a bug in ops.py that constant's definition are overlapped.